wolfcrypt/src/port/aria/aria-crypt.c

```diff
@@ -38,6 +38,8 @@ size and a key size of 128, 192, or 256 bits.
 #include <wolfssl/wolfcrypt/error-crypt.h>
 #include <wolfssl/ssl.h>
 #include <wolfssl/wolfcrypt/port/aria/aria-crypt.h>
+#include <wolfssl/wolfcrypt/aes.h>
```

Comment:
It's unusual that this should still be necessary. Have you looked to see what is the difference in Make vs CMake and why the latter needs this?

Reply:
I'll see what I can find. 

Without that `aes.h` include, I see this error when running `cmake --build .`

```
Consolidate compiler generated dependencies of target wolfssl
[  1%] Building C object CMakeFiles/wolfssl.dir/wolfcrypt/src/port/aria/aria-crypt.c.o
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-crypt.c: In function ‘CheckAriaGcmIvSize’:
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-crypt.c:144:21: error: ‘GCM_NONCE_MIN_SZ’ undeclared (first use in this function)
  144 |     return (ivSz == GCM_NONCE_MIN_SZ ||
      |                     ^~~~~~~~~~~~~~~~
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-crypt.c:144:21: note: each undeclared identifier is reported only once for each function it appears in
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-crypt.c:145:21: error: ‘GCM_NONCE_MID_SZ’ undeclared (first use in this function)
  145 |             ivSz == GCM_NONCE_MID_SZ ||
      |                     ^~~~~~~~~~~~~~~~
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-crypt.c:146:21: error: ‘GCM_NONCE_MAX_SZ’ undeclared (first use in this function)
  146 |             ivSz == GCM_NONCE_MAX_SZ);
      |                     ^~~~~~~~~~~~~~~~
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-crypt.c: In function ‘wc_AriaGcmSetIV’:
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-crypt.c:175:42: error: ‘AES_IV_FIXED_SZ’ undeclared (first use in this function)
  175 |         (ivFixed != NULL && ivFixedSz != AES_IV_FIXED_SZ)) {
      |                                          ^~~~~~~~~~~~~~~
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-crypt.c: In function ‘CheckAriaGcmIvSize’:
/mnt/c/test/wolfssl-PR-aria/wolfcrypt/src/port/aria/aria-crypt.c:147:1: error: control reaches end of non-void function [-Werror=return-type]
  147 | }
      | ^
cc1: all warnings being treated as errors
gmake[2]: *** [CMakeFiles/wolfssl.dir/build.make:286: CMakeFiles/wolfssl.dir/wolfcrypt/src/port/aria/aria-crypt.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:101: CMakeFiles/wolfssl.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
gojimmypi:/mnt/c/test/wolfssl-PR-aria/out
```

Reply:
Similar chain of includes as ecc. aria-cryptocb.h -> cryptocb.h -> aes.h

Reply:
@JacobBarthelmeh AND @bandi13 I've confirmed that turning on `OPENSSL_EXTRA` negates the need for the `#include <wolfssl/wolfcrypt/ecc.h>` in `aria-cryptocb.c` and for the `#include <wolfssl/wolfcrypt/aes.h>` & `#include <wolfssl/error-ssl.h>` in  `aria-crypt.c`.

```
cmake ..  -DWOLFSSL_ARIA=yes -DWOLFSSL_OPENSSLEXTRA=yes
```

I've also confirmed the client & server apps work properly without  `OPENSSL_EXTRA` defined.

This also applies to regular `make` build:

```

./configure --enable-aria --disable-opensslall --disable-opensslextra
make
```

Thanks @bandi13 for all the offline comments & suggestions.

I've created https://github.com/wolfSSL/wolfssl/pull/6730 that adds some wolfSSL includes to ARIA and removes the dependency on OpenSSL needing to be enabled to use ARIA.